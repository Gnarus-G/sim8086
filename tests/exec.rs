use std::process::Command;

use insta::assert_display_snapshot;

macro_rules! test_with {
    ($file:literal) => {
        let _ = Command::new("nasm")
            .arg(format!("./fixtures/exec/{}.asm", $file))
            .status()
            .unwrap();

        let app_output = assert_cmd::Command::cargo_bin("sim8086")
            .unwrap()
            .arg(format!("./fixtures/exec/{}", $file))
            .arg("--exec")
            .output()
            .map(|out| String::from_utf8(out.stdout).unwrap())
            .unwrap();

        insta::with_settings!({ description => $file }, {
            assert_display_snapshot!(app_output);
        })
    };
}

#[test]
fn movs() {
    test_with!("immediate_movs");
    test_with!("register_movs");
}

#[test]
fn add_sub_cmp() {
    test_with!("add_sub_cmp");
}

#[test]
fn ip_tracking() {
    test_with!("ip_register");
}

#[test]
fn conditional_jumps() {
    // test_with!("add_sub_cmp");
}
