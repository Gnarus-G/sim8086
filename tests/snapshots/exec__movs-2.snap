---
source: tests/exec.rs
description: register_movs
expression: app_output
---
Registers {
    ax: 0x0004 (4),
    bx: 0x0003 (3),
    cx: 0x0002 (2),
    dx: 0x0001 (1),
    sp: 0x0001 (1),
    bp: 0x0002 (2),
    si: 0x0003 (3),
    di: 0x0004 (4),
    flags: ,
}

